Yank/multistate/multistateanalyzer.py

```diff
@@ -1840,7 +1840,12 @@ def _get_equilibration_data(self, energies=None, replica_state_indices=None):
         # Discard equilibration samples.
         # TODO: if we include u_n[0] (the energy right after minimization) in the equilibration detection,
         # TODO:         then number_equilibrated is 0. Find a better way than just discarding first frame.
-        equilibration_data = list(utils.get_equilibration_data(u_n[1:]))
+        i_t, g_i, n_effective_i = utils.get_equilibration_data_per_sample(u_n[1:])
+        n_effective_max = n_effective_i.max()
+        i_max = n_effective_i.argmax()
+        n_equilibration = i_t[i_max]
+        g_t = g_i[i_max]
+        equilibration_data = [n_equilibration, g_t, n_effective_max]
```

Comment:
Apologies if I'm missing something. What's the difference between using this and keep using `utils.get_equilibration_data()` as before?

Reply:
Good question, that is by design. the `get_equilibration_data_per_sample` is more general than `get_equilibration_data` so I wanted to actually remove `get_equilibration_data` in a future release. To that end, I took all the code which called it and removed its called, but others who use the API may not, so it trows a warning now.